### PR TITLE
Converting String to our Custom Class in C#

### PR DESCRIPTION
**Pull request by WebMat1**
With this extension, you can use signalR with C# as the same way than signalR use.

`SignalRInstance.On<string, int, Type>("methodName", (str, i, type) ={
//Do Something here...
})`

### CODE REVIEW

File: Unity/Assets/Plugins/SignalR/SignalRInterface.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        signalR.AddHandler(methodName);
+    }
+
+    public static void On<T1, T2, T3, T4, T5, T6,T7>(this SignalRInterface signalR, string methodName, Action<T1, T2, T3, T4, T5, T6,T7> handler)
+    {
+        signalR.Functions.TryAdd(methodName, args => handler(args[0].ToObject<T1>(), args[1].ToObject<T2>(), args[2].ToObject<T3>(), args[3].ToObject<T4>(), args[4].ToObject<T5>(), args[5].ToObject<T6>(), args[6].ToObject<T7>()));
+        signalR.AddHandler(methodName);
+    }
+
+    public static void On<T1, T2, T3, T4, T5, T6,T7,T8>(this SignalRInterface signalR, string methodName, Action<T1, T2, T3, T4, T5, T6,T7,T8> handler)
```
Comment (evanlindsey) [inline Unity/Assets/Plugins/SignalR/SignalRInterface.cs:107]: It looks like SignalR has support for up to 7 arguments:
https://github.com/SignalR/SignalR/blob/5efe531359fbe338898745abaaf7894dd28150d9/src/Microsoft.AspNet.SignalR.Client/HubProxyExtensions.cs#L345
For parity, we might also stop at 7.

File: Server/Hubs/MainHub.cs [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
         public async Task SendPayloadA(string payload)
         {
-            var data = JsonSerializer.Deserialize<dynamic>(payload);
-            string json = JsonSerializer.Serialize(data);
+            var msg = "We are in A...";
+            Console.WriteLine(msg);
+            List<object> objects = new List<object>(){msg};
+            string json = JsonSerializer.Serialize(objects);
+            // var data = JsonSerializer.Deserialize<dynamic>(payload);
+            // string json = JsonSerializer.Serialize(data);
```
Comment (evanlindsey) [inline Server/Hubs/MainHub.cs:25]: It looks like this was changed for debugging... should revert before merging to continue using payload.

File: Unity/Assets/Plugins/SignalR/SignalRInterface.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    private void SignalRInterface_HandlerInvoked(object sender, HandlerEventArgs e)
+    {
+        try 
+        { 
+            Action<object[]> action;
+
+            if (Functions.TryGetValue(e.HandlerName, out action))
+            {
+                var arrayParameters = JsonSerializer.Deserialize<List<object>>(e.Payload);
```
Comment (evanlindsey) [inline Unity/Assets/Plugins/SignalR/SignalRInterface.cs:35]: Should first validate that the payload is actually Json (as we want to allow plain Strings).

**Review by evanlindsey — CHANGES_REQUESTED** [review]
@WebMat1 This looks promising! I've left a couple of comments for updates. After merge, I may look to refactor the existing lib class and include this logic.

File: Unity/Assets/Plugins/SignalR/SignalRInterface.cs (hunk 3 above)
Comment (WebMat1) [reply Unity/Assets/Plugins/SignalR/SignalRInterface.cs:35]: Im working on it... Im so sorry about be late, but these days Im a little bit busy...

File: Server/Hubs/MainHub.cs (hunk 2 above)
Comment (WebMat1) [reply Server/Hubs/MainHub.cs:25]: Im working on it... Im so sorry about be late, but these days Im a little bit busy...

**Pull request merged**